lib/info-extras.js

```diff
@@ -124,11 +124,49 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ', ',\n');
+  let watchNextJson;
   try {
     jsonStr = JSON.parse(jsonStr);
-  } catch (err) {
+    watchNextJson = JSON.parse(jsonStr.watch_next_response);
+  }
+  catch (err) {
     return [];
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
-};
+  let rvsParams = jsonStr.rvs.split(',').map((e) => {
+    return {
+      id: util.between(e, 'id=', '&'),
+      shortViewCount: util.between(e, 'short_view_count_text=', '+views')
+    };
+  });
+  let secondaryResults = watchNextJson.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results;
+  let recomendedVideos = [];
+  for (let result of secondaryResults) {
+    let details;
+    if (details = result.compactVideoRenderer) {
+      try {
+        let viewCount = details.viewCountText.simpleText;
+        if(!viewCount)
+          viewCount = details.viewCountText.runs[0].text.replace(' watching', '');
```

Comment:
due to the possibility of this string being different for every language, how about instead split it by spaces and get the first part?
```suggestion
          viewCount = details.viewCountText.runs[0].text.split(' ')[0];
```

Reply:
👍

Reply:
not resolved yet

Reply:
Thank you for the suggestion, resolved